src/features/query-bar/components/TemplateButton.js

```diff
@@ -52,9 +52,14 @@ class TemplateButton extends React.Component<Props, State> {
         let icon;
         let text;
 
-        if (!templates) {
+        const isLoadingTemplates = !templates;
+        const hasTemplates = templates && templates.length === 0;
```

Comment:
It's odd that "hasTemplates" is true when length === 0.

Reply:
Ah this is a typo, should be `hasNoTemplates`